OpenSim/Simulation/Model/Model.cpp

```diff
@@ -661,9 +661,15 @@ void Model::createMultibodyTree()
 
     auto bodies = getComponentList<Body>();
     for (auto& body : bodies) {
-        _multibodyTree.addBody( body.getAbsolutePathName(),
-                                body.getMass(), false,
-                                const_cast<Body*>(&body) );
+        try {
+            _multibodyTree.addBody( body.getAbsolutePathName(),
+                                    body.getMass(), false,
+                                    const_cast<Body*>(&body) );
+        } catch (const std::runtime_error& e) {
+            cout << "Duplicate Body names are not permitted. Please update "
+                 << "your model's .osim file accordingly." << std::endl;
```

Comment:
I've revised to throw a new OpenSim exception as you suggested. The new exception displays the "ensure absolute path names are unique" message and then appends the original exception message verbatim, making no assumption about why the MultibodyGraphMaker actually complained. A problem with my original design is that the `cout` will be displayed even if some other OpenSim code catches and adequately handles the `runtime_error`.